### ISSUE TITLE
Stability issues

### ISSUE THREAD

**Issue by simplenotezy** [juliansteenbakker/mobile_scanner#1098]
First of all I want to thank the authors behind this package. There's defiantly been put a lot of work into it, and it is in my opionion the best library out there, especially because it handles multiple code formats and works well on both platforms.

However, I think it has some major issues. Firstly: currently hot restart is not working. This is a major productivity-hindering issue, as you have to restart the whole application and wait on build time, making development and testing slow (tracked here https://github.com/juliansteenbakker/mobile_scanner/issues/1091 and with PR https://github.com/juliansteenbakker/mobile_scanner/pull/1086 which _partly_ works around the issue).

Additionally, for some reason I find the process of scanning to be flaky. It may work very fast for a few scans, but all of a sudden, it may stop. I've enabled breakpoints for all exceptions, but nothing is outputted, when it won't scan. It seems to be related to using `scanWindowUpdateThreshold: 1000` and dynamically updating the scanWindow with `updateScanWindow`.

I can share code examples and/or screencasts. Initially I just wanted to check if I am the only one experiencing this, and/or if it's already on the roadmap.

**Comment by navaronbracke** [comment]
For the scan window update threshold issue, did you play around with different values? By default we disable the threshold, so the scan window _should_ always update. A threshold of 1000 is quite huge?

We use that to determine the _delta_ of difference between the old and new window

https://github.com/juliansteenbakker/mobile_scanner/blob/master/lib/src/mobile_scanner.dart#L171-L179

**Comment by simplenotezy** [comment]
Created a separate issue for it here: https://github.com/juliansteenbakker/mobile_scanner/issues/1099 - it probably makes more sense to address one issue at a time.

**Comment by simplenotezy** [comment]
I set the treshold to 1000 because when scanning tickets for an event, I found it impossible for any operator to scan tickets faster than every 1 second. Maybe 500ms would be better. But that was the logic I used when determining the update threshold

**Comment by navaronbracke** [comment]
Since we already track the hot reload issue in #1091, perhaps we can close this issue in favor of #1099 ?

**Comment by simplenotezy** [comment]
Yes, good idea! I'll close it
